### PR TITLE
Expand checkstyle

### PR DESCRIPTION
**Pull request by Nightfirecat**
This introduces the charset property to help avoid issues on Windows machines, and adds rules regarding braces (required except for empty-body loops) and import wildcards (disallowed, except static imports).

Depends on !25

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -44,6 +48,9 @@
 			<property name="tokens" value="COMMA"/>
 		</module>
 		<module name="UnusedImports"/>
+		<module name="AvoidStarImport">
+			<property name="allowStaticMemberImports" value="true"/>
```
Comment (deathbeam) [inline checkstyle.xml]: Oh accidentally commented on wrong PR. Is it possible to make this check for > 10 static imports as we have defined in checkstyle?

Comment (Nightfirecat) [reply checkstyle.xml]: We don't have that defined in checkstyle, only in the intellij settings. To my knowledge, there's no way to limit this module to below or above any specific number of imports.

Comment (deathbeam) [reply checkstyle.xml]: Yeah I meant code style: `By changing Names count to use static imports with ''*' to 10 (static imports for f.e. ItemIDs with wildcards are preferred)`. But if it is not possible it is fine I guess

**Comment by Nightfirecat** [comment]
Seems like this isn't wanted.

**Pull request closed without merging**